### PR TITLE
simulator: add optional accel clipping (enabled via parameter)

### PR DESCRIPTION
**Pull request by dagar**
This PR introduces new SIM parameteters that add accelerometer clipping (per axis) when the throttle of any motor is within a certain magic range (73-76%). As we get more raw data from real vehicles using the new IMU drivers I'm hoping this can be improved.

 - SIM_ACCEL_CLIP_X
 - SIM_ACCEL_CLIP_Y
 - SIM_ACCEL_CLIP_Z

### CODE REVIEW

**Comment by dagar** [comment]
Screenshot from 2020-04-06 16-56-57

File: src/modules/simulator/simulator_mavlink.cpp [hunk 1]
```diff
@@ -202,7 +202,49 @@ void Simulator::update_sensors(const hrt_abstime &time, const mavlink_hil_sensor
...
+
+			static constexpr float ACCEL_MAX = CONSTANTS_ONE_G * 16.f;
+
+			int throttle_percentage = round(act.control[actuator_controls_s::INDEX_THROTTLE] * 100);
+
+			if (throttle_percentage >= CLIPPING_MAGIC_PERCENTAGE_MIN && throttle_percentage <= CLIPPING_MAGIC_PERCENTAGE_MAX) {
+
+				// X
+				if (_param_sim_accel_clip_x.get()) {
+					x = ACCEL_MAX * _last_clipping_high[0];
```
Comment (bkueng) [inline src/modules/simulator/simulator_mavlink.cpp:229]: Don't you want to toggle between +-ACCEL_MAX?

Comment (dagar) [reply src/modules/simulator/simulator_mavlink.cpp]: Ha, thanks. Yes that was the idea that was apparently lost somewhere along the way with mindless hacking during a call.

**Comment by julianoes** [comment]
> @julianoes would this fit into the fault triggering you are working on?

@bkueng It could eventually, yes. For now it's a spec only:
https://github.com/mavlink/mavlink/pull/1331

**Pull request closed without merging**